File: uis/backoffice/types/inventory.ts
export type InventoryOrderType = "inbound" | "outbound";

export interface InventoryProduct {
  id: number;
  name: string;
  sku: string;
  stock: number;
  threshold: number;
}

export interface InventoryOrder {
  id: number;
  product_id: number;
  product_name: string;
  quantity: number;
  type: InventoryOrderType;
  notes: string;
  created_at: string;
  created_by: string;
}

export interface InventoryOrderCreate {
  product_id: number;
  quantity: number;
  notes: string;
}

export function isLowStock(product: InventoryProduct): boolean {
  return product.stock < product.threshold;
}

File: services/api/app/models/inventory.py
"""Pydantic models for HealthCore clinic inventory."""

from __future__ import annotations

from enum import Enum

from pydantic import BaseModel, Field


class OrderType(str, Enum):
    inbound = "inbound"
    outbound = "outbound"


class ProductResponse(BaseModel):
    id: int
    name: str
    sku: str
    stock: int
    threshold: int


class OrderCreate(BaseModel):
    product_id: int
    quantity: int = Field(gt=0)
    notes: str = ""


class OrderResponse(BaseModel):
    id: int
    product_id: int
    product_name: str
    quantity: int
    type: OrderType
    notes: str
    created_at: str
    created_by: str
